test/parser/JavaMetrics.test.ts

```diff
@@ -0,0 +1,142 @@
+import { testFileMetrics } from "./TestHelper";
+const javaTestResourcesPath = "./resources/java/";
+
+describe("test metrics calculation for Java language.", () => {
+    describe("parses classes metric", () => {
+        it("count classes with different access-modifier", async () => {
+            await testFileMetrics(javaTestResourcesPath + "Classes.java", "classes", 2);
+        });
+
+        it("count interface, abstract class, enum", async () => {
+            await testFileMetrics(javaTestResourcesPath + "Classlike.java", "classes", 4);
+        });
+
+        it("count nested classes", async () => {
+            await testFileMetrics(javaTestResourcesPath + "NestedClasses.java", "classes", 4);
+        });
+
+        it("count classes with one character as class name", async () => {
+            await testFileMetrics(
+                javaTestResourcesPath + "SingleCharacterClassName.java",
+                "classes",
+                2
+            );
+        });
+
+        it("dont count the fields or methods of class", async () => {
+            await testFileMetrics(
+                javaTestResourcesPath + "ClassWithFieldsAndMethods.java",
+                "classes",
```

Comment:
Let's extract a const to define metric names on a central place once.

Reply:
Done with new FileMetric string enum